dot/node_test.go

```diff
@@ -200,17 +201,75 @@ func TestNewNodeC(t *testing.T) {
 			},
 			err:  errors.New("no keys provided for authority node"),
 		},
-		// TODO this is commented out because in holds a lock on badger db, causing next test to foil
+		{
+			name: "minimal config",
+			args: args{
+				cfg: &Config{
+					Global:  GlobalConfig{BasePath: cfg.Global.BasePath,},
+					Init:    InitConfig{Genesis: genFile.Name()},
+					Account: AccountConfig{Key: "alice"},
+					Core:    CoreConfig{WasmInterpreter: wasmer.Name,},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := NewNodeC(tt.args.cfg)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+				utils.RemoveTestDir(t)
+			} else {
+				assert.NoError(t, err)
+			}
+
+			if tt.want != nil {
+				assert.Equal(t, tt.want.Name, got.Name)
+			}
+		})
+	}
+}
+
+func TestNewNodeMock(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	m := NewMocknewNodeIface(ctrl)
+	m.EXPECT().nodeInitialised(gomock.Any()).Return(true)
+	m.EXPECT().initKeystore(gomock.Any())
+	m.EXPECT().createStateService(gomock.Any())
+	m.EXPECT().createRuntimeStorage(gomock.Any())
+	m.EXPECT().loadRuntime(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any())
+	m.EXPECT().createBlockVerifier(gomock.Any())
+	m.EXPECT().createDigestHandler(gomock.Any())
+	m.EXPECT().createCoreService(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any())
```

Comment:
Please try to not use `gomock.Any()` since it covers code but test depth is quite low and some bugs can sneak in.

Ideally declare both the gomock controller and mocks inside the subtests with parameters from each test case to define the mock behavior.

Reply:
updated to remove use of `gomock.Any()`.